cg/cli/workflow/microsalt/store.py

```diff
@@ -27,101 +30,29 @@ def store(context):
-            context.abort()
-        bundle_obj, version_obj = results
     except FileNotFoundError as error:
         click.echo(click.style(f"missing file: {error.args[0]}", fg="red"))
         context.abort()
 
-    family_obj = _add_new_analysis_to_the_status_api(bundle_obj, status)
-    _reset_action_from_running_on_family(family_obj)
-    new_analysis = _add_new_complete_analysis_record(bundle_data, family_obj, status, version_obj)
-    version_date = version_obj.created_at.date()
-    click.echo(f"new bundle added: {bundle_obj.name}, version {version_date}")
-    _include_files_in_housekeeper(bundle_obj, context, hk_api, version_obj)
-
-    return new_analysis
-
-
-def _include_files_in_housekeeper(bundle_obj, context, hk_api, version_obj):
-    """Function to include files in housekeeper"""
-    try:
-        hk_api.include(version_obj)
-    except hk.VersionIncludedError as error:
-        click.echo(click.style(error.message, fg="red"))
-        context.abort()
-    hk_api.add_commit(bundle_obj, version_obj)
-
-
-def _add_new_complete_analysis_record(bundle_data, family_obj, status, version_obj):
-    """Function to create and return a new analysis database record"""
-    pipeline = family_obj.links[0].sample.data_analysis
-    pipeline = pipeline if pipeline else "mip"  # TODO remove this default from here
-
-    if status.analysis(family=family_obj, started_at=version_obj.created_at):
-        raise AnalysisDuplicationError(
-            f"Analysis object already exists for {family_obj.internal_id}{version_obj.created_at}"
-        )
-
-    new_analysis = status.add_analysis(
-        pipeline=pipeline,
-        version=bundle_data["pipeline_version"],
-        started_at=version_obj.created_at,
-        completed_at=dt.datetime.now(),
-        primary=(len(family_obj.analyses) == 0),
-    )
-    new_analysis.family = family_obj
-    return new_analysis
-
-
-def _reset_action_from_running_on_family(family_obj):
-    family_obj.action = None
-
-
-def _add_new_analysis_to_the_status_api(bundle_obj, status):
-    family_obj = status.family(bundle_obj.name)
-    return family_obj
+    status.add_commit(new_analysis)
+    click.echo(click.style("included files in Housekeeper", fg="green"))
 
 
 @store.command()
 @click.pass_context
 def completed(context):
     """Store all completed analyses."""
-    hk_api = context.obj["hk_api"]
-    for analysis_obj in context.obj["tb_api"].analyses(status="completed", deleted=False):
-        existing_record = hk_api.version(analysis_obj.family, analysis_obj.started_at)
-        if existing_record:
-            LOG.debug("analysis stored: %s - %s", analysis_obj.family, analysis_obj.started_at)
-            continue
-        click.echo(click.style(f"storing family: {analysis_obj.family}", fg="blue"))
-        with Path(analysis_obj.config_path).open() as config_stream:
-            context.invoke(analysis, config_stream=config_stream)
+    pass
```

Comment:
Now that the microbial cases and samples tables are there, I can implement this, but I need some help with this. @patrikgrenfeldt is helping me out with some advice.

Reply:
While I managed to figure out how to get all microbial cases to store, I have some questions for @sylvinite before I can find the correct deliverables files and continue. Since @hassanfa is waiting for this PR to be done, how about we implement this in a later PR? Perhaps we can solve the parsing-the-analysis-date-from-the-file-path thing in the same PR?

Reply:
> Since @hassanfa is waiting for this PR to be done, how about we implement this in a later PR? Perhaps we can solve the parsing-the-analysis-date-from-the-file-path thing in the same PR?

Unfortunately I only have today to work on this (my deadline for it was today), and I won't be able to push this forward for at least the next two-three weeks (need to catch up on other deadlines). You can merge whenever you and codeowners are comfortable with results.

Reply:
@barrystokman I think it is still a good plan to make the "store all completed" in a separate PR and touch base with @sylvinite  on the deliverables file. Would be nice to merge this as it adds nice functionality on its own. Also, @hassanfa can pick up samples from HK, when ready to do so and will no longer be blocked

Reply:
https://github.com/Clinical-Genomics/cg/issues/779 created for store completed, will add new information when that becomes available